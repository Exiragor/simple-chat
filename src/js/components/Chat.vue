<template>
    <div class="row">
            <div class="panel panel-default">
                <div class="panel-heading">Panel heading without title</div>
                <div class="panel-body">
                    <div class="container">
                        <div class="row message-bubble" v-for="msg in messages">
                            <p class="text-muted">Alex</p>
                            <span>{{ msg }}</span>
                        </div>
                    </div>
                    <div class="panel-footer">
                        <div class="input-group">
                            <input type="text" class="form-control" v-model="message">
                            <span class="input-group-btn">
                    <button @click="sendMessage" class="btn btn-default" type="button">Send</button>
                  </span>
                        </div>
                    </div>
                </div>
            </div>
    </div>
</template>

<script>
    import io from 'socket.io-client';

    const socket = io('http://localhost:4050');

    export default {
        name: 'chat',

        data() {
            return {
                messages: [],
                message: "",
            }
        },

        methods: {
            sendMessage() {
                socket.emit('message', this.message);
                this.message = "";
            }
        },

        mounted() {
            let that = this;
            socket.on('message', function(msg){
                that.messages.push(msg);
            });
        }
    }
</script>