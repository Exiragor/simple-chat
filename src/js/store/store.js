import Vue from 'vue';
import Vuex from 'vuex';

Vue.use(Vuex)

export const state = {
    messages: [],
    nickname: {name: '', color: ''},
}

export const actions = {
    changeNickName({commit, state}, nickname) {
        let name = (nickname.name !== undefined) ? nickname.name : state.nickname.name
        let color = (nickname.color !== undefined) ? nickname.color : state.nickname.color
        commit('CHANGE_NICKNAME', { name, color })
    },
    addMessage({commit}, msg) {
        commit('ADD_MESSAGE', msg)
    },
}

export const mutations = {
    CHANGE_NICKNAME(state, nickname) {
        state.nickname.name = nickname.name;
        state.nickname.color = nickname.color;
    },
    ADD_MESSAGE(state, message) {
        state.messages.push(message)
    },
}

export const getters = {
    messages(state) {
        return state.messages
    },
    nickname(state) {
        return state.nickname
    }
}


export default new Vuex.Store({
    state,
    actions,
    mutations,
    getters
});
