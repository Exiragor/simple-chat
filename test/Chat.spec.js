import { shallow } from '@vue/test-utils'
import Chat from '@/components/Chat.vue'
import store from '@/store/store'

describe('Chat.vue', () => {
    it('renders message-bubble for each msg in messages', () => {
        const items = ['', '']
        const wrapper = shallow(Chat, {
            computed: { messages: () => items }
        })
        expect(wrapper.findAll('.message-bubble')).toHaveLength(items.length)
    })

    it('call sendMessage on press button', () => {
        const stub = jest.fn()
        const wrapper = shallow(Chat, {
            computed: { messages: () => [] },
            methods: {sendMessage: stub}
        })

        wrapper.find('button').trigger('click')

        expect(stub).toBeCalled()
    })

    it('call sendMessage on press enter', () => {
        const stub = jest.fn()
        const wrapper = shallow(Chat, {
            computed: { messages: () => [] },
            methods: {sendMessage: stub}
        })

        wrapper.find('input').trigger('keypress', {key: 'Enter'})

        expect(stub).toBeCalled()
    })

    it('matches snapshot', () => {
        const items = [
            { text: 'text 1', author: 'author 1', color: '#ff0000'},
            { text: 'text 2', author: 'author 2', color: '#ff0000'},
        ]
        const wrapper = shallow(Chat, {
            computed: { messages: () => items }
        })

        expect(wrapper.html()).toMatchSnapshot()
    })

    describe('Tests with store', () => {
        const wrapper = shallow(Chat, {
            store,
            data: { text: "test"}
        })
        wrapper.vm.sendMessage()

        it('text returns to empty', () => {
            expect(wrapper.vm.$data.text).toBe('')
        })
        it('sendMessage add a new msg for render', () => {
            expect(wrapper.findAll('.message-bubble')).toHaveLength(1)
        })
    })
});