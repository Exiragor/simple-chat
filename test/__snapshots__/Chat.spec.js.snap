// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat.vue matches snapshot 1`] = `
<div class="row">
  <div class="panel panel-default">
    <div class="panel-heading">Chat</div>
    <div class="panel-body">
      <div class="container">
        <div class="row message-bubble">
          <p class="text-muted"><span style="color: rgb(255, 0, 0);">author 1</span></p> <span>text 1</span></div>
        <div class="row message-bubble">
          <p class="text-muted"><span style="color: rgb(255, 0, 0);">author 2</span></p> <span>text 2</span></div>
      </div>
      <div class="panel-footer">
        <div class="input-group">
          <input type="text" class="form-control"> <span class="input-group-btn"><button type="button" class="btn btn-default">Send</button></span></div>
      </div>
    </div>
  </div>
</div>
`;
